src/main/java/net/coderbot/iris/pipeline/ClearPassCreator.java

```diff
@@ -62,14 +64,32 @@ public static ImmutableList<ClearPass> createClearPasses(RenderTargets renderTar
 				}
 
 				// No need to clear the depth buffer, since we're using Minecraft's depth buffer.
-				clearPasses.add(new ClearPass(clearColor,
-						renderTargets.createFramebufferWritingToAlt(clearBuffers)));
+				clearPasses.add(new ClearPass(clearColor, renderTargets::getCurrentWidth, renderTargets::getCurrentHeight,
+						renderTargets.createFramebufferWritingToAlt(clearBuffers), GL21C.GL_COLOR_BUFFER_BIT));
 
-				clearPasses.add(new ClearPass(clearColor,
-						renderTargets.createFramebufferWritingToMain(clearBuffers)));
+				clearPasses.add(new ClearPass(clearColor, renderTargets::getCurrentWidth, renderTargets::getCurrentHeight,
+						renderTargets.createFramebufferWritingToMain(clearBuffers), GL21C.GL_COLOR_BUFFER_BIT));
 			}
 		});
 
 		return ImmutableList.copyOf(clearPasses);
 	}
+
+	public static ImmutableList<ClearPass> createShadowClearPasses(ShadowRenderTargets renderTargets,
+																   PackShadowDirectives renderTargetDirectives) {
+		List<ClearPass> clearPasses = new ArrayList<>();
+
+		for (int i = 0; i < renderTargets.getNumColorTextures(); i++) {
+			if (i < renderTargetDirectives.getColorSamplingSettings().size()) {
+				if (renderTargetDirectives.getColorSamplingSettings().get(i).getClear()) {
+					PackShadowDirectives.SamplingSettings samplingSettings = renderTargetDirectives.getColorSamplingSettings().get(i);
+					clearPasses.add(new ClearPass(samplingSettings.getClearColor(), renderTargets::getResolution, renderTargets::getResolution, renderTargets.getFramebufferForColorTexture(i), GL21C.GL_COLOR_BUFFER_BIT | GL21C.GL_DEPTH_BUFFER_BIT));
+				} else {
+					clearPasses.add(new ClearPass(new Vector4f(1.0F), renderTargets::getResolution, renderTargets::getResolution, renderTargets.getFramebufferForColorTexture(i), GL21C.GL_DEPTH_BUFFER_BIT));
```

Comment:
This code seems a little bit awkward and sorta locks in the limitation of two shadow color buffers. Either this should be more documented, or the code should be reworked so that it doesn't so strongly lock in the limitation.